### ISSUE TITLE
automatic temperature spacing in ParallelTemperingSampler

### ISSUE THREAD

**Issue by cwalker7** [choderalab/openmmtools#488]
If I specify min_temperature, max_temperature, and n_temperatures when using ParallelTemperingSampler to create a simulation, the temperatures computed in ParallelTemperingSampler here: https://github.com/choderalab/openmmtools/blob/master/openmmtools/multistate/paralleltempering.py#L157-L159 are off. 

```
temperatures = [min_temperature + (max_temperature - min_temperature) *
                            (math.exp(i / n_temperatures-1) - 1.0) / (math.e - 1.0)
                            for i in range(n_temperatures)]  # Python 3 uses true division for /
```

For example, setting min_temperature=200K, max_temperature=300K, n_temperatures=12, we get this unexpected result:

temperatures=[163.21, 165.07, 167.09, 169.29, 171.68, 174.28, 177.10, 180.17, 183.50, 187.13, 191.07, 195.34]

**Comment by jchodera** [comment]
Huh. OK, I can reproduce this locally if I use unit-bearing `min_temperature` and `max_temperature`:
![Uploading image.png…]()

**Comment by jchodera** [comment]
And also if I apparently don't use units!
![image](https://user-images.githubusercontent.com/3656088/97381326-16139b00-1886-11eb-9182-9c3c46560304.png)

**Comment by jchodera** [comment]
Fixed by #489

**Comment by cwalker7** [comment]
Looks good now, thanks for fixing.
